purchase_request_to_requisition/models/purchase_request.py

```diff
@@ -21,36 +21,38 @@ class PurchaseRequestLine(models.Model):
     @api.depends('requisition_lines')
     def _compute_is_editable(self):
         super(PurchaseRequestLine, self)._compute_is_editable()
-        if self.requisition_lines:
-            self.is_editable = False
+        for rec in self:
+            if rec.requisition_lines:
+                rec.is_editable = False
 
     @api.multi
     def _compute_requisition_qty(self):
-        self.ensure_one()
-        requisition_qty = 0.0
-        for requisition_line in self.requisition_lines:
-            if requisition_line.requisition_id.state != 'cancel':
-                requisition_qty += requisition_line.product_qty
-        self.requisition_qty = requisition_qty
+        for rec in self:
+            requisition_qty = 0.0
+            for requisition_line in rec.requisition_lines:
+                if requisition_line.requisition_id.state != 'cancel':
+                    requisition_qty += requisition_line.product_qty
+            rec.requisition_qty = requisition_qty
 
     @api.multi
     @api.depends('requisition_lines.requisition_id.state')
     def _compute_requisition_state(self):
-        self.requisition_state = 'none'
-        if self.requisition_lines:
-            if any([pr_line.requisition_id.state == 'done' for
-                    pr_line in
-                    self.requisition_lines]):
-                self.requisition_state = 'done'
-            elif all([pr_line.requisition_id.state == 'cancel'
-                      for pr_line in self.requisition_lines]):
-                self.requisition_state = 'cancel'
-            elif any([pr_line.requisition_id.state == 'in_progress'
-                      for pr_line in self.requisition_lines]):
-                self.requisition_state = 'in_progress'
-            elif all([pr_line.requisition_id.state in ('draft', 'cancel')
-                      for pr_line in self.requisition_lines]):
-                self.requisition_state = 'draft'
+        for rec in self:
+            rec.requisition_state = 'none'
```

Comment:
knowing that `recordset.fields = 'value'` will  involve a `models.write()` I think it will be more clean to use a temporary variable that contains the `requisition_state` and then make the `self.requisition_state = temp` in the end. What do you think?